pkg/volume/metadata/metadata_test.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/testclient"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir"
+)
+
+const basePath = "/tmp/fake"
+
+func formatMap(m map[string]string) string {
+	var l string
+	for key, value := range m {
+		l += key + "=" + value + "\n"
+	}
+	return l
+}
+
+func sortStringAndCompare(lhs, rhs string) bool {
+	lhsSplitted := strings.Split(lhs, "")
+	sort.Strings(lhsSplitted)
+	rhsSplitted := strings.Split(rhs, "")
+	sort.Strings(rhsSplitted)
+	return strings.Join(rhsSplitted, "") != strings.Join(lhsSplitted, "")
+}
+
+func newTestHost(t *testing.T, client client.Interface) volume.VolumeHost {
+	tempDir, err := ioutil.TempDir(basePath, "metadata_volume_test.")
+	if err != nil {
+		t.Fatalf("can't make a temp rootdir: %v", err)
+	}
+	return volume.NewFakeVolumeHost(tempDir, client, empty_dir.ProbeVolumePlugins())
+}
+
+func TestCanSupport(t *testing.T) {
+	pluginMgr := volume.VolumePluginMgr{}
+	pluginMgr.InitPlugins(ProbeVolumePlugins(), newTestHost(t, nil))
+
+	plugin, err := pluginMgr.FindPluginByName(metadataPluginName)
+	if err != nil {
+		t.Errorf("Can't find the plugin by name")
+	}
+	if plugin.Name() != metadataPluginName {
+		t.Errorf("Wrong name: %s", plugin.Name())
+	}
+}
+
+func TestLabels(t *testing.T) {
```

Comment:
If we're going to support pod name and namespace in the volume plugin we probably need tests for those as well.